src/gui/callstatechecker.cpp

```diff
@@ -94,6 +96,13 @@ void CallStateChecker::reset()
     setChecking(true);
 }
 
+void CallStateChecker::slotReceivedPushNotification(Account *account)
+{
+    if (account->id() == _accountState->account()->id()) {
+        startCallStateCheck();
+    }
+}
```

Comment:
should we not rate limit this ?
like not more often than the period of `_statusCheckTimer` timer
@nickvergessen any hint ? I would prefer not to create huge server load just because the desktop client try to be smart

Reply:
Originally I wanted to make it so that in this PR the `callstatechecker` would not poll the server but only check for a push notification and then check, in my testing however my server did not always send a push notification on joining/cancellation of the call... Looking at the iOS talk code and in my testing it also seems that that app uses the polling method too

In its current state I agree we should rate limit, though I would like to know if I missed something and we can do it the way I originally intended since that should greatly ease the load on the server

Reply:
https://github.com/nextcloud/notify_push/blob/main/lib/Listener.php#L123
It seems that the same type is used for new notifications as for deleting notifications.
So I guess the debounce might kick in and prevent the removing to be pushed or something? cc @icewind1991

Reply:
But yeah given the situation polling might be the best, but then again for our company call that basically means roughly 70 requests more every 5 seconds until everyone joined.

Reply:
It would be fairly trivial to change this PR to only check the call state once the push notification PR, and we already have code in this class to check for the server version if this is something that will require a patch there 

@mgallien I can make these changes and we can keep this PR open until the required server-side changes (if there are required changes) are made, and then we can merge if that is ok

Reply:
@mgallien @nickvergessen I have changed the code to *only* check the call state when a push notification is received, but only if the server version is a theoretical `24.0.3`. 

Once the situation changes we can change this theoretical server version to whichever real version includes the changes, and then merge this. Until then we can keep the PR open I would say